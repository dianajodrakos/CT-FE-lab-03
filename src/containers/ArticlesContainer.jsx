import React, { Component } from 'react';
import { fetchArticles } from '../services/newsAPI';
import ArticleList from '../components/articles/ArticleList';
import Controls from '../components/articles/Controls';
import Header from '../components/header/Header';

export default class NewsContainer extends Component {
  state = {
    loading: true,
    articles: [],
    query: ''
  }

  async componentDidMount() {
    const articles = await fetchArticles();
    this.setState({
      articles,
      loading: false,
    });
  }

  handleInputChange = (event) => {
    this.setState({ query: event.target.value });
  }

  handleSubmit = async (event) => {
    event.preventDefault();

    this.setState({ loading: true });

    const articles = await fetchArticles(this.state.query);
    this.setState({ articles, loading: false });
  }

  render() {
    console.log(this.state);
    const { articles, loading, query } = this.state;
    return (
      <>
        <Header />

        { loading 
          ? <p>Loading...</p>
          : <main>
            <Controls 
              query={query} 
              onChange={this.handleInputChange} 
              onSubmit={this.handleSubmit}
            />
            <ArticleList articles={articles} /> 
          </main>
        }
            
      </>
    );
  }
}
