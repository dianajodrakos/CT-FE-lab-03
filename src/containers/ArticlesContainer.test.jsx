
import dotenv from 'dotenv';
dotenv.config();
import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import ArticlesContainer from './ArticlesContainer.jsx';

describe('ArticlesContainer', () => {
  it('should display a list of news articles', async () => {
    render(<ArticlesContainer />);

    screen.getByText('Reading Makes You Smarter');

    const ul = await screen.findByRole(
      'list', 
      { name: 'articles' }, 
      { timeout: 2000 }
    );
    expect(ul).not.toBeEmptyDOMElement();

    const input = await screen.findByLabelText('Search Articles:');
    userEvent.type(input, 'Tesla');

    const submitButton = await screen.findByRole(
      'button', 
      { name: 'query-articles' }
    );
    userEvent.click(submitButton);

    return waitFor(() => {
      const articles = screen.findAllByText('Tesla', { exact: false });
      expect(articles).toMatchSnapshot();
    }, { timeout: 2000 });
  });
});
