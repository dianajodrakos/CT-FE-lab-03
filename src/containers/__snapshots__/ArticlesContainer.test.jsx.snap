// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlesContainer should display a list of news articles 1`] = `Promise {}`;
